src/machine.rs

```diff
@@ -113,6 +116,17 @@ impl MemoryExtra {
                     .extern_statics
                     .insert("__cxa_thread_atexit_impl", place.ptr.assert_ptr().alloc_id)
                     .unwrap_none();
+
+                // "environ"
+                let layout = this.layout_of(this.tcx.types.usize)?;
+                let place = this.allocate(layout, MiriMemoryKind::Machine.into());
+                this.write_scalar(Scalar::from_machine_usize(0, &*this.tcx), place.into())?;
+                this.memory
+                    .extra
+                    .extern_statics
+                    .insert("environ", place.ptr.assert_ptr().alloc_id)
```

Comment:
Please rebase, you'll need to intern this as a symbol here similar to what we do with `__cxa_thread_atexit_impl`.